File: back-end/src/main/java/com/bandweaver/maxsec/entity/Alarm.java
package com.bandweaver.maxsec.entity;

import lombok.Data;

import java.io.Serializable;
import java.util.Date;

/**
 * 告警
 */
@Data
public class Alarm implements Serializable {
    private Integer id;
    /**
     * 生成时间
     */
    private Date time;
    /**
     * 告警类型
     * 周界告警1-999
     * 视频告警1000-1999
     * 视频服务告警2000-2999
     * 可视对讲告警3000-3999
     * 高后果区告警4000-4999
     * 火灾消防告警5000-5999
     */
    private Integer alarmTypeId;
    /**
     * 监测对象Id
     */
    private Integer objId;
    /**
     * 告警等级
     */
    private Integer level;
    /**
     * 告警描述
     */
    private String description;
    /**
     * 是否清除
     */
    private Boolean cleaned;
    /**
     * 清除时间
     */
    private Date cleanedTime;
    /**
     * 告警源
     */
    private String source;

    /**
     * 时间戳
     * @return
     */
    public long getTimeStamp(){
        return this.getTime().getTime();
    }
}

File: front-end/src/api/alarm.ts
import request from '@/util/request'
import {
    AlarmVo,
    Alarm
} from '@/types/alarm.interface'

export function getDefaultAlarm(): Alarm {
    return {
        id: 0,
        alarmTypeId: 0,
        objId: 0,
        level: 0,
        description: "",
        cleaned: false,
        source: '',
        time: new Date(),
        alarmType: {
            id: 0,
            name: "",
            objectType: 0,
            objectTypeName: '',
            description: '',
        },
        objectName: '',
        levelName: '',
    }
}

/** 获取告警统计信息 */
export function getAlarmStatistics() {
    return request({
        url: 'alarm_type_statistic',
        method: 'get'
    })
}

/** 获取某场站告警统计信息 */
export function getAlarmStatisticsByStation(stationId: number) {
    return request({
        url: `stations/${stationId}/alarm_type_statistic`,
        method: 'get'
    })
}

/** 获取告警趋势 */
export function getAlarmTrend() {
    return request({
        url: 'alarm_type_trend',
        method: 'get'
    })
}

/** 根据条件分页搜索告警 */
export function getPageAlarmsByCondition(data: AlarmVo) {
    return request({
        url: 'alarms/condition',
        method: 'post',
        data: data
    })
}

/** 根据条件获取告警数目 */
export function getCountByCondition(vo: AlarmVo) {
    return request({
        url: 'alarm_count/condition',
        method: 'post',
        data: vo
    })
}

/** 获取告警的图片地址 */
export function getImagePathOfAlarm(id: number) {
    return request({
        url: `alarms/${id}/images`,
        method: 'get'
    })
}

/** 获取告警的视频地址 */
export function getVideoPathOfAlarm(id: number) {
    return request({
        url: `alarms/${id}/videos`,
        method: 'get'
    })
}

/** 告警清除 */
export function cleanAlarm(id: number) {
    return request({
        url: `alarms/${id}/clean`,
        method: "get"
    })
}